pkg/data/collector.go

```diff
@@ -114,20 +114,24 @@ func (c *Collector) collectFromRelay(ctx context.Context, relay *builder.Client)
 		case <-ctx.Done():
 			return
 		case slot := <-slots:
-			payload, err := c.collectBidFromRelay(ctx, relay, slot)
-			if err != nil {
-				logger.Warnw("could not get bid from relay", "error", err, "relayPublicKey", relayID, "slot", slot)
-				// TODO implement some retry logic...
-				continue
-			}
-			if payload == nil {
-				// No bid for this slot, continue
-				// TODO consider trying again...
-				continue
+			// Querying for 3 bids per slot
+			for i := 0; i < 3; i++{
```

Comment:
This could work, there are 2 minor issues I see though;

1. If theres an error in either `collectBidFromRelay` of if the `payload == nil` then the next loop will happen immediately.  Instead of queries being spaced apart they would all be bunched together. 
2. If all 3 queries succeed they will all be in the first 300 ms of the slot. Would it make sense to get bids later in the slot of would validators have to have proposed a block already?

One other possible but unlikely issue in most cases would be that the sleep pushes us into the next slot but there are still iterations left in the loop so we end up missing data points for 1 slot and getting extra for another. Might pop up in the first slot the monitor checks but probably not after that. 

I think this works if if we're not too worried about the errors causing immediate retries and we want the bids within the first 300 ms. Otherwise I think we could do something a bit more slot aware in the clock to generate multiple ticks per slot.

Reply:
You're right, I've modified the sleep time so its dynamic with the time left until the next slot.